src/main/webapp/app/assessment/assessment-locks/assessment-locks.component.html

```diff
@@ -41,18 +41,7 @@ <h2>
                     </td>
                     <td>
                         <span *jhiHasAnyAuthority="['ROLE_ADMIN', 'ROLE_INSTRUCTOR', 'ROLE_EDITOR', 'ROLE_TA']">
-                            <a
-                                [routerLink]="[
-                                    '/course-management',
-                                    courseId,
-                                    submission.participation!.exercise!.type + '-exercises',
-                                    submission.participation!.exercise!.id!,
-                                    'submissions',
-                                    submission.id,
-                                    'assessment'
-                                ]"
-                                class="btn btn-outline-secondary btn-sm mb-1"
-                            >
+                            <a [routerLink]="getAssessmentLink(courseId, submission)" class="btn btn-outline-secondary btn-sm mb-1">
```

Comment:
I am not sure if this is a better solution, to be honest since method calls in the template code should be avoided.

Reply:
This comment is representative of all occurrences.

Reply:
that is true, but the assessment routes differ a lot depending on the type of exercise, so using a static string doesnt cut it here, also we now have a single method that always returns that route (and can be found via usages) instead of static occurences all over the code ^^